crimson/star_fusion.py

```diff
@@ -41,70 +109,209 @@
 }
 
 
-def parse_lr_entry(lr_gene: str, lr_brkpoint: str) -> dict:
+def parse_lr_entry(
+    break_side: str,
+    entries: Dict[str, str]
+) -> Dict[str, Union[str, int]]:
     """Parse the gene and breakpoint entry.
 
-    :param lr_gene: Column value for 'LeftGene' or 'RightGene'.
-    :param lr_brkpoint: Column value for 'LeftBreakpoint' or 'RightBreakpoint'.
+    :param break_side: The side of the break, right or left
+    :param entires: The entries from the current line
 
     """
-    lrgname, lrgid = lr_gene.split(_DELIM["gids"])
-    lrchrom, lrpos, lrstrand = lr_brkpoint.split(_DELIM["loc"])
-
-    return {
-        "geneName": lrgname,
-        "geneID": lrgid,
-        "chromosome": lrchrom,
-        "position": int(lrpos),
-        "strand": lrstrand,
-    }
+    if break_side == "left":
+        gene = entries["LeftGene"]
+        breakpoint = entries["LeftBreakpoint"]
+        prefix = "Left"
+    elif break_side == "right":
+        gene = entries["RightGene"]
+        breakpoint = entries["RightBreakpoint"]
+        prefix = "Right"
+    else:
+        raise RuntimeError("Please specify either right or left")
+    gname, gid = gene.split(_DELIM["gids"])
+    chrom, pos, strand = breakpoint.split(_DELIM["loc"])
+
+    breakpoint_side = {
+        "geneName": gname,
+        "geneID": gid,
+        "chromosome": chrom,
+        "position": int(pos),
+        "strand": strand,
+    }  # type: Dict[str, Union[str, int]]
+
+    # Get the other side-specific fields from the output
+    sided_fields = [field for field in entries if field.startswith(prefix)]
+
+    # Remove the ones we already parsed
+    if prefix == "Left":
+        sided_fields.remove("LeftGene")
+        sided_fields.remove("LeftBreakpoint")
+    else:  # prefix is either "Left or "Right":
+        sided_fields.remove("RightGene")
+        sided_fields.remove("RightBreakpoint")
```

Comment:
What do you think about combining this with the list comprehension above & the camel-casing functionality below?

Something like:

```
def camelize(field: str, prefix: str) -> str:
    new_field = field[len(prefix):]
    return new_field[0].lower() + new_field[1:]

sided_fields = {
    camelize(field): field_value
    for field, field_value in entries.items()
    if (
        field.startswith(prefix)
        and field not in {f"{prefix}Gene", f"{prefix}Breakpoint"}
    )
}

breakpoint_side.update(**sided_fields)

return breakpoint_side
```